### DISCUSSION TITLE
chrome.runtime is undefined in content script

### DISCUSSION THREAD

**Discussion by pengyancdl** [mubaidr/vite-vue3-chrome-extension-v3#80] [Q&A] [answered]
I added the following into the index.ts in the src/content-script/index.ts file, and got error: chrome.runtime is undefined. What am I missing?

window.addEventListener('messageFromAnotherVueProject', (event) => {
  const info = event.detail;
  chrome.runtime.sendMessage({ action: 'goForIt', data: info });
});

**Comment by mubaidr** [comment]
I just tested, and see no problem accessing `runtime` in content script. Can you please share more details? like code or repo?

![Capture](https://github.com/user-attachments/assets/39f2cdf9-da68-4c55-8847-78596b917896)

**Comment by pengyancdl** [comment, marked as answer]
I finally figured out, you cannot call index.ts more than one time. Otherwise, the 2nd time or 3rd time will cause the problem.

**Reply by mubaidr** [reply]
Glad to hear!

**Comment by mubaidr** [comment]
We're developing a new and improved browser extension template with exciting features like authentication, payment processing, and more. Your feedback is crucial to us!

## [Please take a moment to fill out our feedback form](https://forms.gle/2nzS2AQkVGmqHTLV6)  and let us know if you would be interested in purchasing template, along with any features you would like to see.

Thank you for your time and support!
